### ISSUE TITLE
[BUG] MQTT timeout

### ISSUE THREAD

**Issue by xefil** [thanksmister/android-mqtt-alarm-panel#182]
**Describe the bug**
Hello,
Well, first of all, great software. Exactly what I was searching for!
I'm noticing a strange behaviour, which I don't know where it's related. If MQTT Broker site or on the App or on the Tablet. Well, All went well, until I leave the Tablet some minutes in idle. The wifi connection remains, because I'm pinging the device continuosly. BTW, if I try to alarm, nothing happens. **AND** if I go into settings and immediatly back, then a connection to mqtt broker is done again and the command is sent out. Seems like the command stay in queue, a new session to the broker is made and the commands are sent. Here the broker logs:

```
2021-08-30T14:09:59: Client XeFireAlarmPanel already connected, closing old connection.
2021-08-30T14:09:59: New client connected from 192.168.3.136:39510 as XeFireAlarmPanel (p2, c0, k60, u'mqtt_user').
2021-08-30T14:09:59: Sending CONNACK to XeFireAlarmPanel (1, 0)
2021-08-30T14:09:59: Received PUBLISH from XeFireAlarmPanel (d0, q1, r1, m1, 'alarmpanel/sensor/face', ... (15 bytes))
2021-08-30T14:09:59: Sending PUBACK to XeFireAlarmPanel (m1, rc0)
2021-08-30T14:09:59: Received PUBLISH from XeFireAlarmPanel (d0, q1, r1, m2, 'alarmpanel/sensor/motion', ... (15 bytes))
2021-08-30T14:09:59: Sending PUBACK to XeFireAlarmPanel (m2, rc0)
2021-08-30T14:09:59: Received SUBSCRIBE from XeFireAlarmPanel
2021-08-30T14:09:59: XeFireAlarmPanel 0 alarmpanel/command
2021-08-30T14:09:59: XeFireAlarmPanel 0 home/alarm
2021-08-30T14:09:59: XeFireAlarmPanel 0 home/alarm/event
2021-08-30T14:09:59: Sending SUBACK to XeFireAlarmPanel
2021-08-30T14:09:59: Sending PUBLISH to XeFireAlarmPanel (d0, q0, r1, m0, 'home/alarm', ... (8 bytes))
2021-08-30T14:10:07: Received PUBLISH from XeFireAlarmPanel (d0, q1, r1, m4, 'alarmpanel/state', ... (68 bytes))
2021-08-30T14:10:07: Sending PUBACK to XeFireAlarmPanel (m4, rc0)
2021-08-30T14:10:08: Received PUBLISH from XeFireAlarmPanel (d0, q1, r1, m5, 'home/alarm/set', ... (8 bytes))
2021-08-30T14:10:08: Sending PUBACK to XeFireAlarmPanel (m5, rc0)
2021-08-30T14:10:08: Sending PUBLISH to XeFireAlarmPanel (d0, q0, r0, m0, 'home/alarm/event', ... (34 bytes))
2021-08-30T14:10:08: Sending PUBLISH to XeFireAlarmPanel (d0, q0, r0, m0, 'home/alarm', ... (6 bytes))
```
If you look at the time, you can see that in the same second at the beginnig, the new session is opened and the commands sent out. I've pressed the ARM button before I was in the settings. Seems settings force a new MQTT session and the commands are sent. Possible there are missing `PINGREQ/PINGRESP` that holds the session open? Or is there someting other I could look?
Mqtt broker version below.

**To Reproduce**
Look above.

**Smartphone (please complete the following information):**
 - Device: Fire Tablet
 - OS: Fire OS 7 - 7th gen
 - Application version: 1.1.2 Build 0
 - Mqtt Broker: mosquitto 2.0.10 on docker

Thanks a lot!

Simon

**Comment by thanksmister** [comment]
Seems that the MQTT server is either crashing or shutting down.   I have the same Fire Tablet and OS but do not have this issue.   In my setup, I am Home Assistant MQTT integration as the broker.  I have the built-in screensaver feature on to keep the application always "on" and active.    There are times, maybe once a month, where the tablet just decides to put the application into the background and then the device in regular sleep mode.    I know you are pinging the WiFi, but maybe try using the screensaver to always keep the app in the forefront?

**Comment by thanksmister** [comment]
Going into the settings will shut down the MQTT server, and going back out of the settings will restart it, that's how it's designed. It actually shuts down and starts the background service which handles the camera and MQTT server connection.

**Comment by xefil** [comment]
Hello, 
Thanks for the fast answer. 
I'll try to replace mqtt server moving from mosquitto to rabbitmq tomorrow and let you know. 
Before that I'll try the built in Screensaver and let you know but as far I remember I've tried all display options. 
Thanks for the help :-) 
My fire OS was "rooted" with xda fire toolkit removing whatever was possible. 
Cheers, 
Simon

Il giorno 30 Ago 2021, 16:30, alle ore 16:30, Mister ***@***.***> ha scritto:

**Comment by xefil** [comment]
Hello @thanksmister ,
No good news. I've tried to switch from mosquitto to rabbitmq + mqtt plugin.
Same result.
I was in:
- tcpdump on the docker server on which is mqtt server installed
- mosquitto_sub to see the content of the messages
- checking the logs
After some time the commands from the panel are not taken into consideration. so only if i go to settings and back to the panel, to force a reconnect of the client, then the commands are sent out.
Do you have an idea how to debug further?
Next step is to try Mqtt server on a raspberry to bypass the docker stack, but would avoid, due in this mode I've all on a NAS and not on a poor RPi.
Thanks!
Simon

**Comment by thanksmister** [comment]
Did you see I added a new APK to the release section that has a different MQTT library.  It’s actually for newer Android 9 devices that had permission issues with background running services.  Feel free to give it a try, it only supports back to Android 21 though.   

For me it sounds like the network or service is dying and disconnects the MQTT client.  Going to it the settings and out basically restarts the client again, so something is killing the background service.

Memory issues?  Wifi disconnected?

**Comment by thanksmister** [comment]
It’s also possible your broker is dropping the connection. Maybe it’s in the logs.

**Comment by xefil** [comment]
Hello @thanksmister , no logs into Mqtt at all, even if in debug mode.... but...
I've moved the mqtt broker to a different pc, always with docker, but based on Ubuntu. And moved back to mosquitto, which is surely lighter instead of RabbitMQ. It's working definitely better. Your app, updated to latest, works like a charm, fast and reactive. FYI I've tested your updated app on the old broker with same issues, it means your app is ok! So the question is, how many mqtt messages I was loosing 'til today?
Well the Synology is better it does only the storage...
I think we can close this case, if you haven't any other question.
Simon
